File: frameworks/spring-boot/ADSMediaController.java
package com.adsmedia.sdk;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.Map;

/**
 * Spring Boot REST Controller for ADSMedia
 */
@RestController
@RequestMapping("/email")
public class ADSMediaController {

    @Autowired
    private ADSMediaClient adsMediaClient;

    @PostMapping("/send")
    public ResponseEntity<Map<String, Object>> sendEmail(@RequestBody ADSMediaClient.SendEmailRequest request) {
        try {
            Map<String, Object> result = adsMediaClient.send(request);
            return ResponseEntity.ok(result);
        } catch (Exception e) {
            return ResponseEntity.internalServerError()
                    .body(Map.of("error", e.getMessage()));
        }
    }

    @PostMapping("/batch")
    public ResponseEntity<Map<String, Object>> sendBatch(@RequestBody ADSMediaClient.BatchEmailRequest request) {
        try {
            Map<String, Object> result = adsMediaClient.sendBatch(request);
            return ResponseEntity.ok(result);
        } catch (Exception e) {
            return ResponseEntity.internalServerError()
                    .body(Map.of("error", e.getMessage()));
        }
    }

    @GetMapping("/check")
    public ResponseEntity<Map<String, Object>> checkSuppression(@RequestParam String email) {
        try {
            Map<String, Object> result = adsMediaClient.checkSuppression(email);
            return ResponseEntity.ok(result);
        } catch (Exception e) {
            return ResponseEntity.internalServerError()
                    .body(Map.of("error", e.getMessage()));
        }
    }

    @GetMapping("/ping")
    public ResponseEntity<Map<String, Object>> ping() {
        try {
            Map<String, Object> result = adsMediaClient.ping();
            return ResponseEntity.ok(result);
        } catch (Exception e) {
            return ResponseEntity.internalServerError()
                    .body(Map.of("error", e.getMessage()));
        }
    }
}

File: frameworks/aspnet-core/ADSMediaController.cs
using Microsoft.AspNetCore.Mvc;

namespace ADSMedia
{
    /// <summary>
    /// ASP.NET Core Controller for ADSMedia
    /// </summary>
    [ApiController]
    [Route("api/email")]
    public class ADSMediaController : ControllerBase
    {
        private readonly ADSMediaClient _client;

        public ADSMediaController(ADSMediaClient client)
        {
            _client = client;
        }

        [HttpPost("send")]
        public async Task<IActionResult> Send([FromBody] SendEmailRequest request)
        {
            try
            {
                var result = await _client.SendAsync(request);
                return Ok(result);
            }
            catch (Exception ex)
            {
                return StatusCode(500, new { error = ex.Message });
            }
        }

        [HttpPost("batch")]
        public async Task<IActionResult> SendBatch([FromBody] BatchEmailRequest request)
        {
            try
            {
                var result = await _client.SendBatchAsync(request);
                return Ok(result);
            }
            catch (Exception ex)
            {
                return StatusCode(500, new { error = ex.Message });
            }
        }

        [HttpGet("check")]
        public async Task<IActionResult> CheckSuppression([FromQuery] string email)
        {
            try
            {
                var result = await _client.CheckSuppressionAsync(email);
                return Ok(result);
            }
            catch (Exception ex)
            {
                return StatusCode(500, new { error = ex.Message });
            }
        }

        [HttpGet("ping")]
        public async Task<IActionResult> Ping()
        {
            try
            {
                var result = await _client.PingAsync();
                return Ok(result);
            }
            catch (Exception ex)
            {
                return StatusCode(500, new { error = ex.Message });
            }
        }
    }
}

